internal/tfeprovider/team_access.go

```diff
@@ -0,0 +1,30 @@
+package tfeprovider
+
+type TeamAccess struct {
+	ForEach     map[string]TeamAccess  `json:"for_each,omitempty"`
+	TeamID      string                 `json:"team_id"`
+	WorkspaceID string                 `json:"workspace_id"`
+	Access      string                 `json:"access,omitempty"`
+	Permissions *TeamAccessPermissions `json:"permissions,omitempty"`
+
+	// TODO: Allow resources to support dynamic attrs via embedded structs or methods
+	// Avoid these awkwardly named structs/exposed implementation details
+	DynamicPermissions *DynamicPermissions `json:"dynamic,omitempty"`
```

Comment:
where do you see the line between fully dynamic resources vs typing each attribute? For example, in a workspace resource, would it be a good idea to add a field that renders an arbitrary `map[string]interface{}`? Would help for adopting new resource attributes before we've had time to update the action itself.

Reply:
`dynamic` is hard, that's what my comment refers to, rather than arbitrary attrs. Not sure where I was going with this suggestion, it's a tricky one. You could implement `json.Marshaler` to customize the output. Hard to generalize how to marry `dynamic` with code generation in another language.

As for arbitrary attributes, I think no. Little point in going to all the effort to deal in typed provider resource objects only to leave open an untyped escape hatch. In this action specifically, you need inputs for new workspace attributes anyway.